### PR TITLE
LINGI2263: Add second summary, updated for 2020--2021

### PR DESCRIPTION
**Pull request by Peiffap**

### CODE REVIEW

File: src/q7/cl-INGI2263/summary2/chapters/cm34.tex [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+\gamma(h) = \sum_{\{w \in \mathcal{W}: C(h, w) > 0\}} \frac{d_C}{C(h)},
+\]
+and should be redistributed over the unobserved \(N\)-grams.
+
+\paragraph{Katz Backoff Smoothing} One way of redistributing \(\gamma(h)\) is given by the \textbf{Katz backoff smoothing}:
+\[
+\Prhat(w \mid h) = \left\{\begin{array}{l@{\quad}l}
+\frac{C(h, w) - d_C}{C(h)}, & \textnormal{if} \quad C(h, w) > 0,\\
+\frac{1}{Z} \gamma(h) \Prhat_{\textnormal{backoff}}(w \mid h), & \textnormal{if} \quad C(h, w) = 0,
+\end{array}\right.
```
Comment (Jimvy) [inline src/q7/cl-INGI2263/summary2/chapters/cm34.tex:154]: Just wondering, is there a reason for not using `\begin{cases} ... \end{cases}` ?

**Review by Jimvy — APPROVED** [review]
LGTM 🙂, with minor points of discussion.

In cm8 and cm9, in the algorithm environments, I have issues with a "font shape T1/lmr/m/scit undefined" warning. It looks like the `\Return` and `\eIf` commands put their content in an italic environment, which means that when calling `\textsc` it tries to find a font shape for small-caps + italic, which is not available, and so it instead uses small-caps + slanted.
I don't know if it's possible to remove the auto italic in the commands, or to disable this kind of error altogether, but for now I suggest putting the small-caps in a `\textnormal` so that it is really small-caps+normal.

I would pretty much like this issue to be fixed before merging, so that we reduce the number of warnings down to just the usual "You have requested document class", "Token not allowed in a PDF string", and "Command \underbar has changed".

Also, there is an overfull `\hbox` somewhere, it can be quickly fixed.

I don't have read the document so I don't know if there are any mistake though, but I guess not ;)

File: src/q7/cl-INGI2263/summary2/chapters/cm1.tex [hunk 2]
```diff
@@ -0,0 +1,26 @@
+\section{Introduction}
+\begin{mydef}
+	\textbf{Computational linguistics}, or \textbf{natural language processing}, can be described as the combination between two separate topics:
+	\begin{itemize}
+		\item \textbf{linguistics}: the scientific study of languages, and
+		\item \textbf{computation/processing}: the use of computers or automated processing methods.
+	\end{itemize}
+\end{mydef}
+Figure~\ref{fig:typical_tasks} gives a summary of typical NLP tasks.
+\begin{figure}[!hbtp]
```
[the remark this answers is not in the document]
Comment (Peiffap) [reply src/q7/cl-INGI2263/summary2/chapters/cm1.tex:10]: I used to put `H` as default until I read somewhere that it was bad practice; I've now switched to putting `!hbtp` everywhere. In all honesty, I'm not sure what it does, but it seems to give good results so I see no reason to change.

File: src/q7/cl-INGI2263/summary2/chapters/cm2.tex [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+In \textbf{syntactic analysis}, the goal is to build a \emph{parse tree} of a sentence.
+These parse trees represent the full \emph{structure} and the \emph{connection} between the different parts of a sentence.
+
+\subsubsection{Semantic Analysis}
+Building parse trees is potentially \emph{ambiguous} on a semantical level.
+\textbf{Semantic analysis} allows us to distinguish between syntactically different (but correct) intepretations of a same sentence.
+
+\subsubsection{Higher Levels of Analysis}
+\begin{itemize}
+	\item \textbf{Discourse analysis} is concerned with the analysis of text/speech beyond single sentences/utterances, by looking at \emph{relationships} between various sentences and paragraphs.
```
[the remark this answers is not in the document]
Comment (Peiffap) [reply src/q7/cl-INGI2263/summary2/chapters/cm2.tex]: I went with the second solution (in all honesty, partly to have a paper trail if I ever need to solve that problem again). Fyi, the correct hyphenation options for "utterance" are "ut-ter-ance."

**Comment by Peiffap** [comment]
Thanks for the review @Jimvy; it's a good thing indeed to keep warnings to a minimum. I replied to your comments above. Let me know if there's anything else you want me to update.

> I don't have read the document so I don't know if there are any mistake though, but I guess not ;)

That seems optimistic; I'm sure there are typos and other minor mistakes left. We'll have to hope some of the newer generations are selfless enough to correct them.

File: src/q7/cl-INGI2263/summary2/chapters/cm2.tex (hunk 3 above)
[the remark this answers is not in the document]
Comment (Jimvy) [reply src/q7/cl-INGI2263/summary2/chapters/cm2.tex]: Good thing to know indeed ^^

Sorry for the bad hyphenation of "utterances", I didn't look it up. Another advantage of the second solution: it doesn't rely on human correctness.

File: src/q7/cl-INGI2263/summary2/chapters/cm1.tex (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jimvy) [reply src/q7/cl-INGI2263/summary2/chapters/cm1.tex:10]: I think that means "put it here, or on the bottom, or on the top, or on a dedicated page for figures as a last resort, but I strongly prefer you to put it here and respect my order of preferences".

`H` means "put it here, even if it causes the figure to go out of the page, global warming to accelerate, or Trump to be elected again".

File: src/q7/cl-INGI2263/summary2/chapters/cm2.tex [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+	A \textbf{part-of-speech} is a \emph{grammatical category} identifying the \emph{nature} and/or \emph{syntactic function} of a word.
+\end{mydef}
+
+\subsubsection{Syntactic Analysis}
+In \textbf{syntactic analysis}, the goal is to build a \emph{parse tree} of a sentence.
+These parse trees represent the full \emph{structure} and the \emph{connection} between the different parts of a sentence.
+
+\subsubsection{Semantic Analysis}
+Building parse trees is potentially \emph{ambiguous} on a semantical level.
+\textbf{Semantic analysis} allows us to distinguish between syntactically different (but correct) intepretations of a same sentence.
```
Comment (Jimvy) [inline src/q7/cl-INGI2263/summary2/chapters/cm2.tex:48]:
```suggestion
\textbf{Semantic analysis} allows us to distinguish between syntactically different (but correct) interpretations of a same sentence.
```

**Review by Jimvy — COMMENTED** [review]
I found one 😁

**Comment by Jimvy** [comment]
One less PR

**Pull request merged**